src/game/game.js

```diff
@@ -21,33 +28,73 @@ class Game {
         handle: () => this.world.getPlayer().moveRight(),
       },
     };
-
-    this.world = new World();
-    this.render = new Render(canvas, this.world, this.inputController);
   }
 
   createGame() {
+    this.world = new World();
+    this.render = new Render(
+      this.canvas,
+      this.world,
+      this.inputController,
+      this.controllerCallback,
+    );
     this.world.generate();
     this.render.animate();
   }
 
+  destroyGame() {
+    this.render = undefined;
+    this.world = undefined;
+    this.removeEventListeners();
+  }
+
   startGame() {
     this.addEventListeners();
   }
 
+  pauseGame() {
+    this.removeEventListeners();
+    this.render.pause();
+  }
+
+  resumeGame() {
+    this.render.resume();
+    this.addEventListeners();
+  }
+
+  restartGame() {
+    this.removeEventListeners();
+    this.world.removeObjects();
+    this.world.generate();
+    this.addEventListeners();
+    this.render.restart();
+  }
+
+  handleInputKeyDown(event) {
+    if (event.key === 'Escape') {
+      this.controllerCallback('pause');
+    }
+    if (this.inputController[event.key]) {
```

Comment:
I would write this as an `else if`. We don't need to check other keys once the user pressed escape.